src/IO/ReadHelpers.h

```diff
@@ -1895,6 +1898,7 @@ bool tryReadQuotedField(String & s, ReadBuffer & buf);
 void readJSONField(String & s, ReadBuffer & buf);
 bool tryReadJSONField(String & s, ReadBuffer & buf);
 
+template<bool support_crlf>
 void readTSVField(String & s, ReadBuffer & buf);
```

Comment:
Let's make 2 non-template functions `readTSVField` and `readTSVFieldCRLF` and make template `readTSVFieldImpl` implementation in .cpp file. It will look much better imho